packages/flutter_tools/lib/src/aot.dart

```diff
@@ -52,102 +74,46 @@ class AotBuilder {
-        }
-
-        // Generate AOT snapshot and compile to arch-specific App.framework.
-        final Map<DarwinArch, Future<int>> exitCodes = <DarwinArch, Future<int>>{};
-        iosBuilds.forEach((DarwinArch iosArch, String outputPath) {
-          exitCodes[iosArch] = snapshotter.build(
-            platform: platform,
-            darwinArch: iosArch,
-            buildMode: buildMode,
-            mainPath: kernelOut,
-            packagesPath: PackageMap.globalPackagesPath,
-            outputPath: outputPath,
-            extraGenSnapshotOptions: extraGenSnapshotOptions,
-            bitcode: bitcode,
-            quiet: quiet,
-            splitDebugInfo: null,
-            dartObfuscation: false,
-          ).then<int>((int buildExitCode) {
-            return buildExitCode;
-          });
-        });
-
-        // Merge arch-specific App.frameworks into a multi-arch App.framework.
-        if ((await Future.wait<int>(exitCodes.values)).every((int buildExitCode) => buildExitCode == 0)) {
-          final Iterable<String> dylibs = iosBuilds.values.map<String>(
-              (String outputDir) => globals.fs.path.join(outputDir, 'App.framework', 'App'));
-          globals.fs.directory(globals.fs.path.join(outputPath, 'App.framework')).createSync();
-          await processUtils.run(
-            <String>[
-              'lipo',
-              ...dylibs,
-              '-create',
-              '-output', globals.fs.path.join(outputPath, 'App.framework', 'App'),
-            ],
-            throwOnError: true,
-          );
-        } else {
-          status?.cancel();
-          exitCodes.forEach((DarwinArch iosArch, Future<int> exitCodeFuture) async {
-            final int buildExitCode = await exitCodeFuture;
-            globals.printError('Snapshotting ($iosArch) exited with non-zero exit code: $buildExitCode');
-          });
-        }
-      } else {
-        // Android AOT snapshot.
-        final int snapshotExitCode = await snapshotter.build(
-          platform: platform,
-          buildMode: buildMode,
-          mainPath: kernelOut,
-          packagesPath: PackageMap.globalPackagesPath,
-          outputPath: outputPath,
-          extraGenSnapshotOptions: extraGenSnapshotOptions,
-          bitcode: false,
-          splitDebugInfo: null,
-          dartObfuscation: false,
-        );
-        if (snapshotExitCode != 0) {
-          status?.cancel();
-          throwToolExit('Snapshotting exited with non-zero exit code: $snapshotExitCode');
-        }
+    if (!result.success) {
+      for (final ExceptionMeasurement measurement in result.exceptions.values) {
+        globals.printError(measurement.exception.toString());
       }
-    } on ProcessException catch (error) {
-      // Catch the String exceptions thrown from the `runSync` methods below.
-      status?.cancel();
-      globals.printError(error.toString());
-      return;
+      throwToolExit(null);
```

Comment:
A message here would probably be helpful, even if it was just something like "The AOT build failed".

Reply:
Done